### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by madushan1000** [mozilla/libdweb#109]
I'm wondering if this project is dead since there are no commits/fixes in some time now and the laters firefox 74.0a1 nightly seems to break the experimental APIs. My addon built using libdweb works fine with firefox 73.0a1 nightly.

**Comment by sammacbeth** [comment]
Which APIs are you using? I believe as of current Firefox nightly you also need to set the pref `extensions.experiments.enabled` to true for the APIs to be loaded.

Regarding the project status I believe the upstreaming of the protocol handler API is currently stalled in review. Maybe @Gozala knows more...

**Comment by Gozala** [comment]
Given that I am no longer a Mozilla employ, I can only speculate what the future of this project might be. However as @sammacbeth pointed out all the APIs were rewritten for landing to Firefox, and were stalled on review. If I do get those reviewed (which seems unlikely) I'll be willing to volunteer to get across the finish line.

**Comment by madushan1000** [comment]
@sammacbeth thank you for the tip, I'll try setting that flag!
Meanwhile, I would love to see the libdweb apis land in the mainline Firefox(hopefully on mobile too). I'm only using the protocol handler api for now, but I would need to use the TCPsocket and TCPlistner apis too in the future. 
@Gozala do you know who else we can reach out to get the changes reviewed?
Thank you very much for the help!

**Comment by sammacbeth** [comment]
For the protocol handler there are some minor fixes in my fork for nightly: https://github.com/sammacbeth/libdweb/commit/aca62bf8cb3e7c337b3ab348c7bcebea8f5f716e

@Gozala I would very much like to see this move forward. If I can do anything on my end please let me know. I'd be happy to lend some dev time to this if you're not able to commit time.

**Comment by Gozala** [comment]
I am afraid I do not know how to move this forward. Asking on corresponding bugzilla issues might help but also may not.

If I'm fully honest I am skeptical Mozilla merging any of these into the tree. Lack of interest/support made it impossible while I was employed. People who cared [are gone](https://techcrunch.com/2020/01/15/mozilla-lays-off-70-as-it-waits-for-subscription-products-to-generate-revenue/) so I have hard time imagining it becoming a priority.

**Comment by Gozala** [comment]
@sammacbeth I've forked this repo to https://github.com/libdweb/libdweb/, none of us have write access here so it's probably best to land any changes there (you should have access to do so).

I have diverged from code here quite a bit in patches attached to bugzilla based on review feedback and the fact that JS streams became a thing in Firefox. It might be worth to try and get those changes incorporated.

I personally can't justify spending more of my time on this. As I don't see a path to landing and I  should really be looking for payed job instead.

**Comment by sammacbeth** [comment]
@Gozala Fair enough, and understandable. I'll push my changes to the libdweb/libdweb repo, and try to incorporate your changes from bugzilla there too. Best of luck for whatever is next for you!
